### PR TITLE
Implement TypedArray#slice

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/typedarray.ts [hunk 1]
```diff
@@ -1235,14 +1279,35 @@ function SORT<TArray extends ArrayBufferView, T>(
...
+function SLICE<TArray extends ArrayBufferView, T>(
+  array: TArray,
+  start: i32,
+  end: i32
+): TArray {
+  var length = array.length;
+  start  = start < 0 ? max(start + length, 0) : min(start, length);
+  end    = end   < 0 ? max(end   + length, 0) : min(end  , length);
+  length = max(end - start, 0);
+  var slice = changetype<TArray>(__alloc(offsetof<TArray>(), idof<TArray>()));
```
Comment (dcodeIO) [inline std/assembly/typedarray.ts]: Isn't this missing a new buffer (.data, .dataStart) as the target of the memory.copy below?

Comment (MaxGraey) [reply std/assembly/typedarray.ts]: I replace this with `instantiate<T>`

**Comment by MaxGraey** [comment]
@dcodeIO Could you re-review this?

File: std/assembly/typedarray.ts [hunk 2]
```diff
@@ -1235,14 +1279,34 @@ function SORT<TArray extends ArrayBufferView, T>(
...
+): TArray {
+  var length = array.length;
+  start  = start < 0 ? max(start + length, 0) : min(start, length);
+  end    = end   < 0 ? max(end   + length, 0) : min(end  , length);
+  length = max(end - start, 0);
+  var slice = instantiate<TArray>(length);
+  memory.copy(
+    slice.dataStart,
+    array.dataStart + (<usize>start << alignof<T>()),
+    length << alignof<T>()
```
Comment (dcodeIO) [inline std/assembly/typedarray.ts]: `<usize>length` to be sure

Comment (MaxGraey) [reply std/assembly/typedarray.ts]: Done

**Comment by MaxGraey** [comment]
@dcodeIO Could you re-review this?

**Comment by dcodeIO** [comment]
Thanks!

**Pull request merged**